source/transfer_list.rst

```diff
@@ -657,5 +658,58 @@ such that the last ACPI table in this entry ends at offset
      - hdr_size
      - One or more ACPI tables.
 
+
+.. _tpm_evlog_entry:
+
+TPM info table entry layout (XFERLIST_TPM_INFO)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds TPM info of the platform.
+
+.. _tab_tpm_info:
+.. list-table:: TPM info type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **5**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - sizeof (base_address) + sizeof (evlog_size) + sizeof (crb_base_address) + sizeof (crb_size)
+
+   * - evlog_base_address
+     - 8
+     - 8
+     - The base address of event log
+
+   * - evlog_size
+     - 0x4
+     - 16
+     - Size of the event log
+
+   * - crb_base_address
+     - 8
+     - 8
```

Comment:
These offset numbers don't make sense, you can't have both the `evlog_base_address` and `crb_base_address` at offset 8. (Also, I think you should write all offsets in hex like the other entries.)

When you correct the offsets I would recommend to make sure all fields are naturally aligned, so since you have a couple of 8-byte and 4-byte fields you probably want to arrange them evlog_base_address, crb_base_address, evlog_size, crb_size.

Reply:
Agree with defining naturally aligned fields.  If we add a new "need_to_replay" boolean as described in an earlier comment we could use that to align the CRB info.